### PR TITLE
Added the hid api.

### PR DESCRIPTION
**Pull request by minorgal**
Add a new adapter to support the web hid api so that we can support USB along with Bluetooth.
Needed to make some small modifications to the base class to allow for different packet sizes.

### CODE REVIEW

File: src/godirect.js [hunk 1]
```diff
@@ -9,12 +10,12 @@ const godirect = {
   * @param {config} config
   * @returns {Promise} Promise object represents a Device instance
   */
-  async createDevice(bleDevice, { open = true, startMeasurements = true } = {}) {
-    let adapter = bleDevice;
+  async createDevice(baseDevice, { open = true, startMeasurements = true, bluetooth = true } = {}) {
+    let adapter = baseDevice;
 
     // If not a go direct adapter, assume a web bluetooth device
```
Comment (jordanaustin) [inline src/godirect.js:16]: Can you update this comment. It should say something like: `// If not a godirect adapter, assume web device`

File: src/godirect.js [hunk 2]
```diff
@@ -35,20 +36,40 @@ const godirect = {
...
   * This can only be invoked via a user interaction (e.g. within a click event) otherwise you'll get a security warning.
   * @name selectDevice
+  * @param {bool} bluetooth
   * @returns {Promise} Promise object represents a Device instance
   */
-  async selectDevice() {
-    if (!navigator.bluetooth) {
-      return Promise.reject(new Error('No Web Bluetooth support.'));
-    }
+  async selectDevice(bluetooth = true) {
```
Comment (jordanaustin) [inline src/godirect.js:42]: This means we're limiting it to only bluetooth or usb devices. I think the selection UI could allow us to display both types, right?

**Comment by minorgal** [comment]
I am ready for final review, if you want to take a gander. 
I had to mimic some functionality that I think will come/change as the webhid develops. 
So we will likely have to revisit the implementation as time goes on.

File: src/godirect.js [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
+  async createDevice(nativeDevice, { open = true, startMeasurements = true } = {}) {
+    let adapter = nativeDevice;
 
-    // If not a go direct adapter, assume a web bluetooth device
+    // If not a go direct adapter then create it based on what was passed in
     if (!adapter.godirectAdapter) {
-      adapter = new WebBluetoothDeviceAdapter(bleDevice);
+      if (nativeDevice.gatt) {
+        adapter = new WebBluetoothDeviceAdapter(nativeDevice);
+      } else if (nativeDevice.collections[0].outputReports[0]) {
```
Comment (jordanaustin) [inline src/godirect.js:20]: you may want to ensure that nativeDevice.collections exists, but I'm not sure if you expect that it always will: 
`nativeDevice.collections && nativeDevice.collections[0].outputReports[0]`

**Pull request merged**